src/roswire/ros2/source.py

```diff
@@ -29,24 +26,32 @@ def for_app_instance(
     ) -> "ROS2PackageSourceExtractor":
         return ROS2PackageSourceExtractor(files=app_instance.files)
 
-    def extract_source_for_package(
+    def get_cmake_info(
         self,
-        path_to_package: str
-    ) -> t.Mapping[str, NodeSourceInfo]:
+        package: Package,
+    ) -> CMakeInfo:
+        path_to_package = package.path
         cmakelists_path = os.path.join(path_to_package, "CMakeLists.txt")
 
         if self._files.isfile(cmakelists_path):
             contents = self._files.read(cmakelists_path)
-            source_infos = extract_sources_from_cmake(contents)
-            return {n.node_name: n for n in source_infos}
+            return self._process_cmake_contents(contents, package, {})
 
         setuppy_path = os.path.join(path_to_package, "setup.py")
         if self._files.isfile(setuppy_path):
             logger.error(
                 "Do not know how to process ROS2 packages with setup.py yet."
             )
-            return {}
+            raise NotImplementedError("Do not know how to process ROS2 "
+                                      "packages with setup.py yet.")
 
         logger.error(f"There is no package information inside "
                      f"{path_to_package}. Is it a package soure directory?")
-        return {}
+        raise ValueError(f"No pacakge information for {path_to_package}.")
+
+    def package_paths(self, package: Package) -> t.Set[str]:
+        # TODO Do this properly
+        include: str = os.path.normpath(
```

Comment:
what is this used for?

Reply:
I think this takes paths with ../../. and gets rid of them.